### DISCUSSION TITLE
Importing / Mounting with the read-only option set

### DISCUSSION THREAD

**Discussion by rgmiller** [openzfs/zfs#11548] [General]
Hello All,

I have a question about importing and mounting with the read-only option set.  Specifically, if I set that option, does that guarantee that absolutely nothing will be written to the devices?  I'm assuming no user-visible changes would be made, but what about things like ZFS's internal state and data structures?  Might those get flushed to disk at some point, such as during an unmount or export operation?  

Thank everyone,
-Ross

**Comment by behlendorf** [comment]
That's right.  When importing a pool read-only all of the vdevs are opened only with read permissions so nothing can be modified.
